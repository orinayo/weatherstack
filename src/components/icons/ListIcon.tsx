import React from 'react'
import {SvgIcon} from './SvgIcon'

export const ListIcon: SvgIcon = ({
  width = 16,
  height = 16,
  strokeColor = 'currentColr',
  strokeWidth = 20,
  fillColor = 'none',
  classes = 'inline-block align-text-bottom mr-1',
}) => (
  <svg
    width={width}
    height={height}
    fill={fillColor}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 512 512"
    className={`stroke-current ${strokeColor} ${classes}`}
  >
    <title>List</title>
    <path
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={strokeWidth}
      d="M160 144h288M160 256h288M160 368h288"
    />
    <circle
      cx="80"
      cy="144"
      r="16"
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={strokeWidth}
    />
    <circle
      cx="80"
      cy="256"
      r="16"
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={strokeWidth}
    />
    <circle
      cx="80"
      cy="368"
      r="16"
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={strokeWidth}
    />
  </svg>
)
