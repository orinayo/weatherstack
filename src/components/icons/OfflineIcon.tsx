import React from 'react'
import { SvgIcon } from './SvgIcon';

const OfflineIcon: SvgIcon = ({
  width = 20,
  height = 20,
  strokeColor = '#ffc200',
  strokeWidth = 1.2,
  fillColor = 'none',
  classes = '',
}) => (
  <svg
    width={width}
    height={height}
    fill={fillColor}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 512 512"
    className={`stroke-current ${strokeColor} ${classes}`}
  >
    <path
      d="M93.72 183.25C49.49 198.05 16 233.1 16 288c0 66 54 112 120 112h184.37M467.82 377.74C485.24 363.3 496 341.61 496 312c0-59.82-53-85.76-96-88-8.89-89.54-71-144-144-144-26.16 0-48.79 6.93-67.6 18.14"
      strokeWidth={strokeWidth}
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      strokeWidth={strokeWidth}
      strokeLinecap="round"
      strokeLinejoin="round"
      d="M448 448L64 64"
    />
  </svg>
)

export default OfflineIcon