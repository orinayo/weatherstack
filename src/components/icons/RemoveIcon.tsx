import React from 'react'
import {SvgIcon} from './SvgIcon'

export const RemoveIcon: SvgIcon = ({
  width = 20,
  height = 20,
  strokeColor = 'text-white',
  strokeWidth = 20,
  fillColor = 'red',
  classes = '',
}) => (
  <svg
    width={width}
    height={height}
    fill={fillColor}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 512 512"
    className={`stroke-current ${strokeColor} ${classes}`}
  >
    <title>Close</title>
    <path
      d="M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z"
      strokeWidth={strokeWidth}
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      strokeWidth={strokeWidth}
      strokeLinecap="round"
      strokeLinejoin="round"
      d="M320 320L192 192M192 320l128-128"
    />
  </svg>
)
