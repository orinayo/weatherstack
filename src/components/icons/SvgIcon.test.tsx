import React from 'react'
import {render} from '@testing-library/react'
import {HeartIcon} from './HeartIcon'
import {ListIcon} from './ListIcon'
import {OfflineIcon} from './OfflineIcon'
import {RemoveIcon} from './RemoveIcon'
describe('SVG Icons', () => {
  test('renders heart svg icon', () => {
    const {asFragment} = render(<HeartIcon />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('renders list svg icon', () => {
    const {asFragment} = render(<ListIcon />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('renders offline svg icon', () => {
    const {asFragment} = render(<OfflineIcon />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('renders remove svg icon', () => {
    const {asFragment} = render(<RemoveIcon />)
    expect(asFragment()).toMatchSnapshot()
  })
})
