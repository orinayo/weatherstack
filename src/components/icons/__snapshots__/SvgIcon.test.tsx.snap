// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG Icons renders heart svg icon 1`] = `
<DocumentFragment>
  <svg
    class="stroke-current text-red "
    fill="white"
    height="20"
    viewBox="0 0 512 512"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Heart
    </title>
    <path
      d="M352.92 80C288 80 256 144 256 144s-32-64-96.92-64c-52.76 0-94.54 44.14-95.08 96.81-1.1 109.33 86.73 187.08 183 252.42a16 16 0 0018 0c96.26-65.34 184.09-143.09 183-252.42-.54-52.67-42.32-96.81-95.08-96.81z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="20"
    />
  </svg>
</DocumentFragment>
`;

exports[`SVG Icons renders list svg icon 1`] = `
<DocumentFragment>
  <svg
    class="stroke-current currentColr inline-block align-text-bottom mr-1"
    fill="none"
    height="16"
    viewBox="0 0 512 512"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      List
    </title>
    <path
      d="M160 144h288M160 256h288M160 368h288"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="20"
    />
    <circle
      cx="80"
      cy="144"
      fill="none"
      r="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="20"
    />
    <circle
      cx="80"
      cy="256"
      fill="none"
      r="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="20"
    />
    <circle
      cx="80"
      cy="368"
      fill="none"
      r="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="20"
    />
  </svg>
</DocumentFragment>
`;

exports[`SVG Icons renders offline svg icon 1`] = `
<DocumentFragment>
  <svg
    class="stroke-current text-yellow-100 "
    fill="none"
    height="20"
    viewBox="0 0 512 512"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M93.72 183.25C49.49 198.05 16 233.1 16 288c0 66 54 112 120 112h184.37M467.82 377.74C485.24 363.3 496 341.61 496 312c0-59.82-53-85.76-96-88-8.89-89.54-71-144-144-144-26.16 0-48.79 6.93-67.6 18.14"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.2"
    />
    <path
      d="M448 448L64 64"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.2"
    />
  </svg>
</DocumentFragment>
`;

exports[`SVG Icons renders remove svg icon 1`] = `
<DocumentFragment>
  <svg
    class="stroke-current text-white "
    fill="red"
    height="20"
    viewBox="0 0 512 512"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Close
    </title>
    <path
      d="M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="20"
    />
    <path
      d="M320 320L192 192M192 320l128-128"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="20"
    />
  </svg>
</DocumentFragment>
`;

exports[`SVG Icons renders remove svg icon 2`] = `
<DocumentFragment>
  <svg
    class="stroke-current  "
    fill="currentColor"
    height="20"
    viewBox="0 0 512 512"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M384 224v184a40 40 0 01-40 40H104a40 40 0 01-40-40V168a40 40 0 0140-40h167.48"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="32"
    />
    <path
      d="M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z"
    />
  </svg>
</DocumentFragment>
`;
